### PR TITLE
add strict debug impl requirements

### PR DESCRIPTION
**Pull request by prasannavl**
## Description
- Add missing `Debug` impls
- Enforce `missing_debug_implementations`
- Add `#[allow(missing_debug_implementations)]` currently, in places where Debug impls may not apply, or ones which probably need a proper manual Debug impl in the future

## Motivation and Context
`Debug` impl is often required by base cases in Rust (say `Result` for `unwrap`/`unwrap_err`) Currently the lack of `Debug` impl on structs like `Error` ends up causing difficultly in usage.  This lint makes sure we avoid it in the future.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/rustasync/tide/blob/master/.github/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/app.rs [hunk 1]
```diff
@@ -236,7 +237,7 @@ impl<AppData: Send + Sync + 'static> App<AppData> {
 ///
 /// This type is useful only in conjunction with the [`HttpService`] trait,
 /// i.e. for hosting a Tide app within some custom HTTP server.
-#[derive(Clone)]
```
Comment (k-nasa) [inline src/app.rs]: [question]
Is there any problem with deleting Clone here?

Comment (prasannavl) [reply src/app.rs]: Oh no. This is a mistake! Thanks for the catch!

Comment (prasannavl) [reply src/app.rs]: Fixed! Thanks.

**Comment by prasannavl** [comment]
This ended up messy having to add tons of `#[allow]` to satisfy the new cookies PR, working around clippy bugs. But this should do it, for now

**Comment by yoshuawuyts** [comment]
@prasannavl oh no, there's another merge conclict! -- I think it might be because of a docs change I just landed. I think a rebase + merge seems good! :+1:

**Pull request merged**